Paper: CNN Based ToF Image Processing (#625)
scipy-conference/scipy_proceedings: papers/trai/trai.rst

:author: Benjamin Kelenyi
:email: Benjamin.Kelenyi@aut.utcluj.ro
:institution: Technical University of Cluj-Napoca

:author: Levente Tamas
:email: Levente.Tamas@aut.utcluj.ro
:institution: Technical University of Cluj-Napoca
:corresponding:

:author: Andrei Cozma
:email: Andrei.Cozma@analog.com
:institution: Analog Devices International

:bibliography: mybib

:video: https://www.youtube.com/watch?v=kANXhHwFrCo

------------------------------
CNN Based ToF Image Processing
------------------------------

.. class:: abstract

   In this paper a Time of Flight (ToF) camera specific data processing pipeline is presented, followed by real life applications using artificial intelligence.
   These applications include use cases such as gesture recognition, movement direction estimation or physical exercises monitoring. 
   The whole pipeline for the body pose estimation is described in details, starting from generating and training phases to the pose estimation 
   and deployment. The final deployment targets were Nvidia Xavier NX and AGX platforms receiving data from an Analog Devices ToF camera.

.. class:: keywords

   transfer learning, ToF, python

Introduction
------------
   

> Last years the evolution of deep neurThe applications targeted are rangingfrom persond detection, counting, activity analysis to volumetricmeasurements,  mapping  and  navigation  with  mobile  agentsal networks also affected the way in which the Time of Flight (ToF) specific images are processed. The images from the ToF cameras are usually obtained as synchronized depth and infrared (IR) image pairs.

Reviewer: I have a printout of an earlier version that I began marking up. I think some part of this first line has been mixed in with later parts.

```suggestion
In recent years the evolution of deep neural networks has affected the way in which Time of Flight (ToF) images are 
processed. Images from ToF cameras are usually obtained as synchronized depth and infrared (IR) image pairs. 
```

.. figure:: bad_plane_segmentation.png
  :width: 400
  :height: 400
  :scale: 40%
  :align: center
  :alt: Limitations of planar segmentation

  Limitations of planar segmentation :label:`limitations`

Due to these shortcomings, a new method for measuring the volume is studied, 
based on the work by :cite:`sommer2020`. 
Their paper, details an algorithm that uses pointclouds with normals 
computed in each point in order to determine collections of point 
pairs for which their normals satisfy the orthogonality constraint.  
The point pair collections will approximate the orthogonal planes. 
By determining the points contained by each orthogonal plane, 
projections can be made that approximate the intersecting lines 
of the orthogonal planes. By selecting the 3 lines that have 
the edge points closest to each other, volume of a box can be computed.
The advantage of this method is that it allows the computation 
of the volume for multiple box shaped objects. 
The following code snippets show the usage of the 
Sommer's plane determination method to compute the volume.

.. code-block:: python
    :linenos:

>     def compute_volume_ortho(pcd,dmin,dman,votes,seg,thresh):

Reviewer 2: So sorry! One more thing -- in the rendered PDF, some of these code lines are extending beyond the column width:

<img width="390" alt="Screen Shot 2021-07-01 at 5 51 04 AM" src="https://user-images.githubusercontent.com/6868396/124113035-94859580-da30-11eb-9e80-8f5409aa2714.png">

we might need to make a few changes like, e.g.

```python
def compute_volume_ortho(
    pcd, dmin, dman, votes, seg, thresh
):
```

Author: @AlexPop104 please make some abbreviation in these lines. Tx!

Reviewer 2: Thanks!
